plugins/modules/ec2_asg.py

```diff
@@ -805,6 +892,14 @@ def get_launch_object(connection, ec2_connection):
                 for instance_type in instance_types:
                     instance_type_dict = {'InstanceType': instance_type}
                     policy['LaunchTemplate']['Overrides'].append(instance_type_dict)
+            if instances_distribution:
+                instances_distribution_params = dict(
```

Comment:
I personally find this flow/formatting a little harder to follow than it should be.

Tests show that the code works, and I know that this is a valid pythonic way to do this.  As such I'll put my +1 but I'd like to have @jillr's opinion if there's a better way to format this before we merge